test/dataframe.jl

```diff
@@ -2032,4 +2032,17 @@ end
     end
 end
 
+@testset "reverse DataFrame" begin
+    df = DataFrame(a=1:5, b=5:-1:1)
+    @test reverse(df) == DataFrame(a=5:-1:1, b=1:5)
+    @test reverse(DataFrame(a=1, b=1)) == DataFrame(a=1, b=1)
+end
+
+@testset "reverse SubDataFrame" begin
+    df = DataFrame(a=1:10, b=10:-1:1, c=11:20)
+    @test reverse(view(df, 1:3, 1:2)) == DataFrame(a=[3, 2, 1], b=[8, 9, 10])
```

Comment:
Also check that the result is a `DataFrame` (and not a `SubDataFrame`) as @bkamins noted.